bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/internal/serviceregistry/ServiceRegistry.java

```diff
@@ -310,6 +313,15 @@ private List<Class<?>> getHookClass(String className, List<Class<?>> hookTypes)
 		}
 	}
 
+	private void setHookRegistrations(String hookClass, List<ServiceRegistrationImpl<?>> hooks) {
+		hooks = hooks == null || hooks.isEmpty() ? Collections.emptyList() : new ArrayList<>(hooks);
+		frameworkHooks.put(hookClass, hooks);
```

Comment:
```suggestion
		frameworkHooks.put(hookClass, hooks == null ? List.of() : List.copyOf(hooks));
```

Reply:
> I assume you are trolling me to use post Java 8+ methods :)

It seems its too long since I'm writing against Java 8 API :-\

ArrayList seems fine then if we assume its only used internally and we always using it "right".